src/flags.zig

```diff
@@ -168,34 +230,45 @@ fn parse_flag(comptime T: type, comptime flag: []const u8, arg: [:0]const u8) T
         return true;
     }
 
-    const value = parse_flag_value(flag, arg);
+    const value = parse_flag_split_value(flag, arg);
     assert(value.len > 0);
-
-    if (T == []const u8) {
-        return value;
-    }
-    assert(@typeInfo(T) == .Int);
-    return parse_flag_value_int(T, flag, value);
+    return parse_value(T, flag, value);
 }
 
 /// Splits the value part from a `--arg=value` syntax.
-fn parse_flag_value(comptime flag: []const u8, arg: [:0]const u8) [:0]const u8 {
+fn parse_flag_split_value(flag: []const u8, arg: [:0]const u8) [:0]const u8 {
+    assert(flag[0] == '-' and flag[1] == '-');
     assert(std.mem.startsWith(u8, arg, flag));
+
     const value = arg[flag.len..];
-    if (value.len < 2) {
-        fatal("{s}: argument requires a value", .{flag});
+    if (value.len == 0) {
+        fatal("{s}: expected value separator '='", .{flag});
     }
     if (value[0] != '=') {
         fatal(
             "{s}: expected value separator '=', but found '{c}' in '{s}'",
             .{ flag, value[0], arg },
         );
     }
+    if (value.len == 1) fatal("{s}: argument requires a value", .{flag});
     return value[1..];
 }
 
+fn parse_value(comptime T: type, flag: []const u8, value: [:0]const u8) T {
+    comptime assert(T != bool);
+    assert((flag[0] == '-' and flag[1] == '-') or flag[0] == '<');
+    assert(value.len > 0);
+
+    if (T == []const u8 or T == [:0]const u8) return value;
+    if (T == ByteSize) return parse_value_size(flag, value);
+    if (@typeInfo(T) == .Int) return parse_value_int(T, flag, value);
+    comptime unreachable;
+}
+
 /// Parse string value into an integer, providing a nice error message for the user.
-fn parse_flag_value_int(comptime T: type, comptime flag: []const u8, value: [:0]const u8) T {
+fn parse_value_int(comptime T: type, flag: []const u8, value: [:0]const u8) T {
+    assert((flag[0] == '-' and flag[1] == '-') or flag[0] == '<');
+
     return std.fmt.parseInt(T, value, 10) catch |err| {
         fatal("{s}: expected an integer value, but found '{s}' ({s})", .{
```

Comment:
The Overflow error message should convey what limit that is being exceeded (e.g. include the name of the integer type in the message).

Previously `parse_cluster`:

```
error.Overflow => fatal("--cluster: value exceeds a 32-bit unsigned integer", .{}),
```

Reply:
Yeah, good point, let me think about a better message here... I think that `exceeds a 32-bit unsigned integer` is perhaps less then ideal here, as this feels too implementation oriented (as a user, I'd rather not care how many bits do you need). But also for something like `--cluster` its not like we could say something better.